validation.go

```diff
@@ -899,15 +896,32 @@ func (v *subSchema) validateNumber(currentSubSchema *subSchema, value interface{
 
 	// format
 	if currentSubSchema.format != "" {
-		if !FormatCheckers.IsFormat(currentSubSchema.format, float64Value) {
-			result.addInternalError(
-				new(DoesNotMatchFormatError),
-				context,
-				value,
-				ErrorDetails{"format": currentSubSchema.format},
-			)
+		if topSchema.HasFormatCheck(currentSubSchema.format) {
```

Comment:
Hmm, actually confused if I'm able to do that. Seems like `validateCommon` is for something else entirely (i.e. contains for enum and const) See below:

```go
func (v *subSchema) validateCommon(topSchema *Schema, currentSubSchema *subSchema, value interface{}, result *Result, context *JsonContext) {

	if internalLogEnabled {
		internalLog("validateCommon %s", context.String())
		internalLog(" %v", value)
	}

	// const:
	if currentSubSchema._const != nil {
		vString, err := marshalWithoutNumber(value)
		if err != nil {
			result.addInternalError(new(InternalError), context, value, ErrorDetails{"error": err})
		}
		if *vString != *currentSubSchema._const {
			result.addInternalError(new(ConstError),
				context,
				value,
				ErrorDetails{
					"allowed": *currentSubSchema._const,
				},
			)
		}
	}

	// enum:
	if len(currentSubSchema.enum) > 0 {
		has, err := currentSubSchema.ContainsEnum(value)
		if err != nil {
			result.addInternalError(new(InternalError), context, value, ErrorDetails{"error": err})
		}
		if !has {
			result.addInternalError(
				new(EnumError),
				context,
				value,
				ErrorDetails{
					"allowed": strings.Join(currentSubSchema.enum, ", "),
				},
			)
		}
	}

	result.incrementScore()
}

```

Reply:
Most `validate*` functions operate on a specific type. Also most checks work on a specific type (a `maxLength` check on a number won't do anything, that's why it's performed in `validateString`).  `validateCommon` works on any instance and that's why only `const` and `enum` are currently in there, because they are the only checks together with `format` that work on a generic JSON type. According to the JSON Schema spec, `format` should also work on any JSON type, just like `const` and `enum`.